app/test/account/like_test.dart

```diff
@@ -87,12 +87,51 @@ void main() {
       final rs1 = await client2.listPackageLikes();
       expect(rs1.likedPackages.isEmpty, true);
 
-      final rs2 = await client.likePackage('hydrogen');
-      expect(rs2.package, 'hydrogen');
+      final rs2 = await client.likePackage(hydrogen.package.name);
+      expect(rs2.package, hydrogen.package.name);
       expect(rs2.liked, true);
 
       final rs3 = await client2.listPackageLikes();
       expect(rs3.likedPackages.isEmpty, true);
     });
+
+    testWithServices('Package likes property is incremented/decremented.',
+        () async {
+      final client = createPubApiClient(authToken: hansUser.userId);
+      final client2 = createPubApiClient(authToken: joeUser.userId);
+
+      final rs = await client.getPackageLikes(hydrogen.package.name);
+      expect(rs.likes, 0);
+      final rs2 = await client2.getPackageLikes(hydrogen.package.name);
+      expect(rs2.likes, 0);
+
+      await client.likePackage(hydrogen.package.name);
+
+      final rs3 = await client.getPackageLikes(hydrogen.package.name);
+      expect(rs3.likes, 1);
+      final rs4 = await client2.getPackageLikes(hydrogen.package.name);
+      expect(rs4.likes, 1);
+
+      await client2.likePackage(hydrogen.package.name);
+
+      final rs5 = await client.getPackageLikes(hydrogen.package.name);
+      expect(rs5.likes, 2);
+      final rs6 = await client2.getPackageLikes(hydrogen.package.name);
+      expect(rs6.likes, 2);
+
+      await client.unlikePackage(hydrogen.package.name);
+
+      final rs7 = await client.getPackageLikes(hydrogen.package.name);
+      expect(rs7.likes, 1);
+      final rs8 = await client2.getPackageLikes(hydrogen.package.name);
+      expect(rs8.likes, 1);
+
+      await client2.unlikePackage(hydrogen.package.name);
+
+      final rs9 = await client.getPackageLikes(hydrogen.package.name);
+      expect(rs9.likes, 0);
+      final rs10 = await client2.getPackageLikes(hydrogen.package.name);
+      expect(rs10.likes, 0);
+    });
```

Comment:
What if:
 * User likes a package that doesn't exist,
 * User unlikes a package that doesn't exist
 * User unlikes a package the user didn't liked (does this decrement, does it cause an error -- it shouldn't)
 * A client requests the list of likes for a package that doesn't exist
 * A client requests the list of likes for a package without attaching an `authorization` header? (You can call `createPubApiClient()` without an `authToken` to get a client doesn't send an `authorization` header), this is just to check that the API end-point doesn't require a user, since we're just requesting public information.

Note: You can use `expectApiException` to test exceptions that makes it pretty easy :)